### PR TITLE
fix(init-1D): #MVS-117 add classes from neo4j relationships for services

### PR DESCRIPTION
**Pull request by flmartineau**
## Describe your changes

- add classes from neo4j relationships (not only the classes field) for creating services

## Checklist tests

- check if all teachers have an associated service with after 1D initialization

## Issue ticket number and link

https://jira.support-ent.fr/browse/MVS-117

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/viescolaire/service/impl/DefaultInitService.java [hunk 1]
```diff
@@ -278,13 +278,8 @@ public Future<JsonObject> initServices(String structureId, SubjectModel subject)
                                         .stream()
                                         .filter(String.class::isInstance)
                                         .map(String.class::cast)
+                                        .distinct()
                                         .collect(Collectors.toList());
-                                List<String> groupIds = teacher.getJsonArray(Field.GROUPIDS, new JsonArray())
```
Comment (flmartineau) [inline src/main/java/fr/openent/viescolaire/service/impl/DefaultInitService.java:282]: On ne prend plus en compte les groupes dans l'initialisation, uniquement les classes

File: src/main/java/fr/openent/viescolaire/service/impl/DefaultUserService.java [hunk 2]
```diff
@@ -804,17 +803,20 @@ public void getTeachers(String idEtablissement, Handler<Either<String, JsonArray
     @Override
     public Future<JsonArray> getTeachersWithClassIds(String structureId) {
         Promise<JsonArray> promise = Promise.promise();
-        String query =  "MATCH (s:Structure {id: {structureId}}) " +
```
Comment (flmartineau) [inline src/main/java/fr/openent/viescolaire/service/impl/DefaultUserService.java:807]: On remonte les enseignants avec leurs classes. On cumule les données du champ `classes` avec les classes correspondants à la relation `MATCH (u: User)-[:IN]->(p:ProfileGroup)-[:DEPENDS]->(c: Class)-[:BELONGS]->(s: Structure)`

Comment (JinDouang) [reply src/main/java/fr/openent/viescolaire/service/impl/DefaultUserService.java:807]: Il faudrait vérifier où est-ce que cette méthode est utilisé (compétences, viesco, edt, cdt etc...)

Juste s'assurer qu'on ne change pas un truc "majeur"

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by qpic-cgi — APPROVED** [review]

File: src/main/java/fr/openent/viescolaire/service/impl/DefaultUserService.java (hunk 2 above)
Comment (flmartineau) [reply src/main/java/fr/openent/viescolaire/service/impl/DefaultUserService.java:807]: J'ai créé cette méthode pour l'initialisation 1D, donc aucun risque que cela soit utilisé ailleurs 😄

**Pull request merged**